satpy/readers/seviri_l2_grib.py

```diff
@@ -0,0 +1,270 @@
+                    h: Height over surface [m]
+                    ssp_lon: longitude of subsatellite point [deg]
+                    nlines: number of lines
+                    ncols: number of columns
+                    a_name: name of the area
+                    a_desc: description of the area
+                    p_id: id of the projection
+                area_dict:
+                    center_point: coordinate of the center point
+                    north: coodinate of the north limit
+                    east: coodinate of the east limit
+                    west: coodinate of the west limit
+                    south: coodinate of the south limit
+
+        """
+        # Read all projection and area parameters from the message
+        earth_major_axis_in_meters = self._get_from_msg(gid, 'earthMajorAxis') * 1000.0   # [m]
+        earth_minor_axis_in_meters = self._get_from_msg(gid, 'earthMinorAxis') * 1000.0   # [m]
+        nr_in_radius_of_earth = self._get_from_msg(gid, 'NrInRadiusOfEarth')
+        xp_in_grid_lengths = self._get_from_msg(gid, 'XpInGridLengths')
+        h_in_meters = earth_major_axis_in_meters * (nr_in_radius_of_earth - 1.0)   # [m]
+
+        # Create the dictionary with the projection data
+        pdict = {
+            'a': earth_major_axis_in_meters,
+            'b': earth_minor_axis_in_meters,
+            'h': h_in_meters,
+            'ssp_lon': self._ssp_lon,
+            'nlines': self._ncols,
+            'ncols': self._nrows,
+            'a_name': 'geos_seviri',
+            'a_desc': 'Calculated area for SEVIRI L2 GRIB product',
+            'p_id': 'geos',
+        }
+
+        # Compute the dictionary with the area extension
+        area_dict = {
+            'center_point': xp_in_grid_lengths + 0.5,
+            'north': self._nrows,
+            'east': 1,
+            'west': self._ncols,
+            'south': 1,
+        }
+
+        return pdict, area_dict
+
+    def _get_xarray_from_msg(self, gid):
+        """Read the values from the GRIB message and return a DataArray object.
+
+        Args:
+            gid: The ID of the GRIB message.
+
+        Returns:
+            DataArray: The array containing the retrieved values.
+
+        """
+        # Data from GRIB message are read into an Xarray...
+        xarr = xr.DataArray(da.from_array(ec.codes_get_values(
+            gid).reshape(self._nrows, self._ncols), CHUNK_SIZE), dims=('y', 'x'))
+
+        return xarr
+
+    def _get_global_attributes(self):
+        """Create a dictionary of global attributes to be added to all datasets.
+
+        Returns:
+            dict: A dictionary of global attributes.
+                ssp_lon: longitude of subsatellite point
+                sensor: name of sensor
+                platform_name: name of the platform
+
+        """
+        orbital_parameters = {
+            'satellite_nominal_longitude': self._ssp_lon,
+            'projection_longitude': self._ssp_lon
+        }
```

Comment:
Sorry for nitpicking again, but it is possible that `satellite_nominal_longitude` differs from `projection_longitude`. For example at times where Meteosat-08 was positioned at -3.4 degrees east, the image was reprojected to 0 degrees. Which value does `self._ssp_lon` have during this period?

Reply:
the L2 GRIB products only have one longitude in the product and this is the `projection longitude`. with your suggestion `satellite_nominal_longitude` could/should be removed. or?

Reply:
@sfinkens what do you think ?

Reply:
@sjoro @mraspaud In this case I'd vote for removing the `satellite_nominal_longitude`. Then it's transparent to the user that we don't have that information.

Reply:
sounds good.